pkg/scheduler/framework/plugins/noderesources/fit_test.go

```diff
@@ -780,3 +780,80 @@ func TestFitScore(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkTestFitScore(b *testing.B) {
+	tests := []struct {
+		name                 string
+		existingPodsNum      int
+		allNodesNum          int
+		nodeResourcesFitArgs config.NodeResourcesFitArgs
+	}{
+		{
+			name:            "1000nodes/RequestedToCapacityRatio",
+			existingPodsNum: 10000,
+			allNodesNum:     1000,
+			nodeResourcesFitArgs: config.NodeResourcesFitArgs{
+				ScoringStrategy: &config.ScoringStrategy{
+					Type:      config.RequestedToCapacityRatio,
+					Resources: defaultResources,
+					RequestedToCapacityRatio: &config.RequestedToCapacityRatioParam{
+						Shape: []config.UtilizationShapePoint{
+							{Utilization: 0, Score: 10},
+							{Utilization: 100, Score: 0},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:            "1000nodes/MostAllocated",
+			existingPodsNum: 10000,
+			allNodesNum:     1000,
+			nodeResourcesFitArgs: config.NodeResourcesFitArgs{
+				ScoringStrategy: &config.ScoringStrategy{
+					Type:      config.MostAllocated,
+					Resources: defaultResources,
+				},
+			},
+		},
+		{
+			name:            "1000nodes/LeastAllocated",
+			existingPodsNum: 10000,
+			allNodesNum:     1000,
+			nodeResourcesFitArgs: config.NodeResourcesFitArgs{
+				ScoringStrategy: &config.ScoringStrategy{
+					Type:      config.LeastAllocated,
+					Resources: defaultResources,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			existingPods, allNodes := st.MakeNodesAndPods(tt.existingPodsNum, tt.allNodesNum)
+			state := framework.NewCycleState()
+			snapshot := cache.NewSnapshot(existingPods, allNodes)
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			fh, _ := runtime.NewFramework(nil, nil, ctx.Done(), runtime.WithSnapshotSharedLister(snapshot))
+			p, err := NewFit(&tt.nodeResourcesFitArgs, fh, plfeature.Features{})
+			if err != nil {
+				b.Fatalf("unexpected error: %v", err)
+			}
+
+			b.ResetTimer()
+
+			pod := st.MakePod().Req(map[v1.ResourceName]string{"cpu": "1000", "memory": "2000"}).Obj()
+			for i := 0; i < b.N; i++ {
+				for _, n := range allNodes[:0] {
```

Comment:
allNodes[:0] is not right, by the way, if we want to benchmark `Score`, simulating one node with pods settled on it is enough unless you want to formulate test cases for different nodes.

Reply:
And I think simulating pods with explicit resource requests will  be better according to what you changed in this PR..

Reply:
> And I think simulating pods with explicit resource requests will be better according to what you changed in this PR..

Thank you for your advice.

Reply:
In the new commit, I simulate one node and pods with explicit resource requests. 

@kerthcet PTAL.